src/containers/dynamicarray.d

```diff
@@ -7,20 +7,42 @@
 
 module containers.dynamicarray;
 
+private import std.experimental.allocator.mallocator : Mallocator;
+private import std.experimental.allocator.common : stateSize;
+
 /**
  * Array that is able to grow itself when items are appended to it. Uses
- * reference counting to manage memory and malloc/free/realloc for managing its
- * storage.
+ * malloc/free/realloc to manage its storage.
+ *
  * Params:
  *     T = the array element type
  *     supportGC = true if the container should support holding references to
  *         GC-allocated memory.
  */
-struct DynamicArray(T, bool supportGC = true)
+struct DynamicArray(T, Allocator = Mallocator, bool supportGC = true)
 {
 	import std.traits : hasMember;
 	this(this) @disable;
 
+	static if (stateSize!Allocator != 0)
```

Comment:
I tried doing this and the compiler wasn't able to figure out the constructor calls correctly.